app/controllers/conversations_controller.rb

```diff
@@ -21,21 +21,24 @@ def index
       end
     end
 
-    gon.contacts = contacts_data
-
     respond_with do |format|
-      format.html
+      format.html { render "index", locals: {no_contacts: current_user.contacts.empty?} }
       format.json { render json: @visibilities.map(&:conversation), status: 200 }
     end
   end
 
   def create
-    contact_ids = params[:contact_ids]
-
-    # Can't split nil
-    if contact_ids
-      contact_ids = contact_ids.split(',') if contact_ids.is_a? String
-      person_ids = current_user.contacts.where(id: contact_ids).pluck(:person_id)
+    contacts_ids = params["contacts-ids"]
+
+    # Contacts autocomplete does not work the same way on mobile and desktop
+    # Mobile returns contact ids array while desktop returns person id
+    # This will have to be removed when mobile autocomplete is ported to Typeahead
+    if session[:mobile_view] && contacts_ids
+      contacts_ids = contacts_ids.split(",") if contacts_ids.is_a? String
+      person_ids = current_user.contacts.where(id: contacts_ids).pluck(:person_id)
+    elsif contacts_ids
+      contacts_ids = contacts_ids.split(",") if contacts_ids.is_a? String
+      person_ids = current_user.contacts.where(person_id: contacts_ids).pluck(:person_id)
```

Comment:
That's a bit better I guess. Submitting person ids as "contact-ids" (why the dash all of the sudden btw? none of our other parameters does that) is still confusing though, perhaps something like:

```rb
recipients_param = %i(contact_ids person_ids).find {|param| params[param].present? }
person_ids  = current_user.contacts.where(recipients_param => params[recipients_param].split(",")).pluck(:person_id)